### PR TITLE
Reconcile tools versions

### LINKED ISSUE #1906: Consolidate the third party required versions
When we run `make install-tools` to install or update the using tools, part of the required tool versions are in the [requirements.sh](https://github.com/fybrik/fybrik/blob/master/hack/tools/requirements.sh), others are in the [make-rules/tools.mk](https://github.com/fybrik/fybrik/blob/master/hack/make-rules/tools.mk) file.
For example: 
GOBIN=$(ABSTOOLBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@**v0.11.1**

We need to consolidate all required tools and their version in a single file.

### PR DESCRIPTION
**Pull request by aradhalevy**
reconciled the requirements to a single file, `requirements.env`, that used to be `requirements.sh`, and updated the places it used to be `requirements.sh`.

also updated kind version to 0.17.0

/closes #1906 

> note: the `set -o nounset` command that used to be in `requirements.sh` in unnecessary, as it is called in `common.sh` and works as expected
 
Signed-off-by: Arad Halevy <arad.halevy@ibm.com>

### CODE REVIEW

File: samples/OneClickDemo/OneClickDemo-Katalog.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -e
 HELM_VERSION=v3.10.3
 YQ_VERSION=4.6.0
 KUBE_VERSION=1.22.0
-KIND_VERSION=0.14.0
+KIND_VERSION=0.17.0
```
Comment (roytman) [inline samples/OneClickDemo/OneClickDemo-Katalog.sh:9]: can you add a rule to the main Mainfile to automatically update these versions?

File: samples/OneClickDemo/OneClickDemo-OMD.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ set -e
 HELM_VERSION=v3.10.3
 YQ_VERSION=4.6.0
 KUBE_VERSION=1.22.0
-KIND_VERSION=0.14.0
+KIND_VERSION=0.17.0
```
Comment (roytman) [inline samples/OneClickDemo/OneClickDemo-OMD.sh:9]: as above

File: samples/OneClickDemo/OneClickDemo-Katalog.sh (hunk 1 above)
Comment (aradhalevy) [reply samples/OneClickDemo/OneClickDemo-Katalog.sh:9]: hmmm
I don't see a way to automatically update these versions from the Makefile, as when you use OneClickDemo you don't clone the Fybrik repo or use its Makefile, only use `curl` to a script in Fybrik main Github repo, not locally...

I can add a `curl` command inside the script that reads the current updated version from Fybrik master branch if you think that is better

Comment (roytman) [reply samples/OneClickDemo/OneClickDemo-Katalog.sh:9]: No, I thought about a templatefile(s) that can be "compiled" into the final file that will be used by curl. 
We can talk tomorrow.

Comment (aradhalevy) [reply samples/OneClickDemo/OneClickDemo-Katalog.sh:9]: as we discussed, added a make target 'reconcile-requirements'

File: samples/OneClickDemo/OneClickDemo-OMD.sh (hunk 2 above)
Comment (aradhalevy) [reply samples/OneClickDemo/OneClickDemo-OMD.sh:9]: same as above answer

File: Makefile [hunk 3]
```diff
@@ -45,6 +45,13 @@ generate: $(TOOLBIN)/controller-gen $(TOOLBIN)/json-schema-generator
 generate-docs:
 	$(MAKE) -C site generate
 
+.PHONY: reconcile-requirements
+reconcile-requirements:
+	sed -i 's/HELM_VERSION=.*/HELM_VERSION=$(HELM_VERSION)/' $(ROOT_DIR)/samples/OneClickDemo/OneClickDemo-OMD.sh $(ROOT_DIR)/samples/OneClickDemo/OneClickDemo-Katalog.sh
```
[the remark this answers is not in the document]
Comment (roytman) [reply Makefile:50]: I'd add CERT_MANAGER_VERSION and AWSCLI_VERSION to the `requirements.env` too and update both places here and in the site documentation. 
In the end, we will have all versions concentrated in a single place - `requirements.env`

File: hack/tools/requirements.env [hunk 4]
```diff
@@ -1,8 +1,7 @@
-#!/usr/bin/env bash
 # Copyright 2020 The Kubernetes Authors.
 # SPDX-License-Identifier: Apache-2.0
 
-set -o nounset
+# after updating any of these version, please run 'make reconcile-requirements' from the root Makefile
```
Comment (roytman) [inline hack/tools/requirements.env:4]: "after updating any of these version, please run 'make reconcile-requirements' at the root project directory"

as an option?

File: site/docs/get-started/OneClickDemo.md [hunk 5]
```diff
@@ -32,12 +34,12 @@ Alternatively you can try Fybrik with Katalog,  a data catalog stub, strictly us
 
 === "Demo with OpenMetaData" 
     ```bash
-    curl https://raw.githubusercontent.com/fybrik/fybrik/master/samples/OneClickDemo/OneClickDemo-OMD.sh | bash -
+    export FYBRIK_VERSION={{ currentRelease|default('master') }}; curl https://raw.githubusercontent.com/fybrik/fybrik/master/samples/OneClickDemo/OneClickDemo-OMD.sh | bash -
```
Comment (roytman) [inline site/docs/get-started/OneClickDemo.md:37]: you added the FYBRIK_VERSION env variable, but do not use it.
should it be:
```
curl https://raw.githubusercontent.com/fybrik/fybrik/${FYBRIK_VERSION}/samples/OneClickDemo/OneClickDemo-OMD.sh 
```

File: site/docs/get-started/OneClickDemo.md [hunk 6]
```diff
@@ -32,12 +34,12 @@ Alternatively you can try Fybrik with Katalog,  a data catalog stub, strictly us
...
     ```
 
 === "Demo with Katalog" 
     ```bash
-    curl https://raw.githubusercontent.com/fybrik/fybrik/master/samples/OneClickDemo/OneClickDemo-Katalog.sh | bash -
+    export FYBRIK_VERSION={{ currentRelease|default('master') }}; curl https://raw.githubusercontent.com/fybrik/fybrik/master/samples/OneClickDemo/OneClickDemo-Katalog.sh | bash -
```
Comment (roytman) [inline site/docs/get-started/OneClickDemo.md:42]: the same

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (aradhalevy) [reply Makefile]: added CERT_MANAGER_VERSION and AWSCLI_VERSION to the requirements.env.

the `sed `command with added "" does not work on  LINUX,

I replaced it with a `perl` command that should work fine on MAC as well, if you could test it on your MAC i would appreciate it

File: hack/tools/requirements.env (hunk 4 above)
Comment (aradhalevy) [reply hack/tools/requirements.env]: sound great, fixed

File: site/docs/get-started/OneClickDemo.md (hunk 5 above)
Comment (aradhalevy) [reply site/docs/get-started/OneClickDemo.md]: that was not my intention, the variable is used in the script itself,
but your suggestion is good also, added

File: site/docs/get-started/OneClickDemo.md [hunk 7]
```diff
@@ -32,12 +34,12 @@ Alternatively you can try Fybrik with Katalog,  a data catalog stub, strictly us
 
 === "Demo with OpenMetaData" 
     ```bash
-    curl https://raw.githubusercontent.com/fybrik/fybrik/master/samples/OneClickDemo/OneClickDemo-OMD.sh | bash -
+    export FYBRIK_VERSION={{ currentRelease|default('master') }}; curl https://raw.githubusercontent.com/fybrik/fybrik/${FYBRIK_VERSION}/samples/OneClickDemo/OneClickDemo-OMD.sh | bash -
```
Comment (roytman) [inline site/docs/get-started/OneClickDemo.md:37]: it should be:
curl https://raw.githubusercontent.com/fybrik/fybrik/{{ currentRelease|default('master') }}/samples/OneClickDemo/OneClickDemo-OMD.sh

and with Katalog the same

Comment (aradhalevy) [reply site/docs/get-started/OneClickDemo.md]: Fixed

**Review by roytman — APPROVED** [review]

**Pull request merged**